app/javascript/src/hooks/useDebounce.js

```diff
@@ -0,0 +1,21 @@
+function debounce(timer, func, timeout = 300) {
```

Comment:
Neat! I wasn't familiar with the term debounce. What a great concept.

```suggestion
// Make the parameter function delay before its invocation. 
// for example, call on an input field to allow a user to type a bit before saving 
// remember to run the cleanup on unloading
function debounce(timer, func, timeout = 300) {
```